src/node.cc

```diff
@@ -3674,44 +3675,35 @@ static void CheckIfAllowedInEnv(const char* exe, bool is_env,
 //  * argv contains the arguments with node and V8 options filtered out.
 //  * exec_argv contains both node and V8 options and nothing else.
 //  * v8_argv contains argv[0] plus any V8 options
-static void ParseArgs(int* argc,
-                      const char** argv,
-                      int* exec_argc,
-                      const char*** exec_argv,
-                      int* v8_argc,
-                      const char*** v8_argv,
+static void ParseArgs(ProcessArguments* process_arguments,
```

Comment:
How about making the parsed arguments a different data type? This will allow to return a single structure here instead of using out-parameters, and also to avoid the mutation of the `ProcessArguments` structure and the `const_cast` in `node::Init()`.